tests/Doctrine/Migrations/Tests/Metadata/Storage/TableMetadataStorageTest.php

```diff
@@ -53,6 +54,37 @@ public function setUp() : void
         $this->storage = new TableMetadataStorage($this->connection, new AlphabeticalComparator(), $this->config);
     }
 
+    public function testCompleteWillAlwaysCastTimeToInteger() : void
+    {
+        $config     = new TableMetadataStorageConfiguration();
+        $executedAt = new DateTimeImmutable('2010-01-05 10:30:21');
+
+        $connection = $this->getMockBuilder(Connection::class)
+            ->setConstructorArgs([
+                ['pdo' => $this->getSqliteConnection()->getWrappedConnection()],
+                new SqlLiteDirver(),
+            ])
+            ->onlyMethods(['insert'])
+            ->getMock();
+
+        $connection
+            ->expects(self::once())
+            ->method('insert')
+            ->with($config->getTableName(), [
+                $config->getVersionColumnName() => '1230',
+                $config->getExecutedAtColumnName() => $executedAt,
+                $config->getExecutionTimeColumnName() => 31000,
```

Comment:
@goetas I understand your approach. I though about the same, but, if you remove the `(int)` cast, the test still passes. That is why I used `mock` and `assertSame`. Wdyt?

Reply:
If the issue only happens on Google Spanner, there is no way we are going to get a failing test, since we don't have Google Spanner tested in our CI, right?

Reply:
In my PR the test fails if we remove the casting, cause I am using `assertSame`. But SQLite does internal cast, different from Spanner.